server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -1164,6 +1167,13 @@ List<String> getIndexSettingsValidationErrors(final Settings settings, final boo
         if (forbidPrivateIndexSettings) {
             validationErrors.addAll(validatePrivateSettingsNotExplicitlySet(settings, indexScopedSettings));
         }
+        int awarenessAttributes = this.awarenessReplicaBalance.awarenessAttributes();
+        if (awarenessAttributes > 1) {
+            if ((settings.getAsInt(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, INDEX_NUMBER_OF_REPLICAS_SETTING.getDefault(Settings.EMPTY))
+                + 1) % awarenessAttributes != 0) {
+                validationErrors.add("expected total copies needs to be a multiple of total awareness attributes " + awarenessAttributes);
+            }
+        }
```

Comment:
Do we also need to mention which attribute zone/rack etc?

Reply:
Good point. It will add the complexity of returning attribute as well from `awarenessAttributes` function , just for logging purpose. So I prefer not doing it . On seeing the error message with and without the attribute, user would anyways need to revisit awareness attributes values . Do you think this behavior is okay ?